crates/types/src/horizon.rs

```diff
@@ -0,0 +1,57 @@
+use nalgebra::{Isometry3, Point2, Vector2};
+use serde::{Deserialize, Serialize};
+use serialize_hierarchy::SerializeHierarchy;
+
+#[derive(Clone, Copy, Debug, Default, Deserialize, Serialize, SerializeHierarchy)]
+pub struct Horizon {
+    pub left_horizon_y: f32,
+    pub right_horizon_y: f32,
+}
+
+impl Horizon {
```

Comment:
Did you change the implementation despite copying?

Reply:
no, the idea of this PR is to refactor first to ease the review. I did not touch any implementation. Implementation changes are coming in a next pr